commons/src/main/java/io/aiven/kafka/tieredstorage/commons/storage/FileFetcher.java

```diff
@@ -24,13 +24,13 @@ public interface FileFetcher {
      * Fetch file.
      * @param key file key.
      */
-    InputStream fetch(String key) throws IOException;
+    InputStream fetch(String key) throws KeyNotFoundException, IOException;
```

Comment:
Should be also in the deleter?

Reply:
Checking idempotent expectations on RSM: https://issues.apache.org/jira/browse/KAFKA-7739?focusedCommentId=17722747&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17722747